client/src/store/modules/simulation.ts

```diff
@@ -94,10 +97,13 @@ const actions: ActionTree<any, HMI.SimulationParameter[]> = {
     });
   },
 
-  async getModelResults ({ getters }, model): Promise<unknown> {
-    return await Promise.all(
+  async fetchModelResults ({ state, getters, dispatch }, { model, aggregator }): Promise<void> {
+    const modelResults: SimulationResponse[] = await Promise.all(
       getters.getSimParameterArray.map(simParamArr => getModelResult(model, simParamArr)),
     );
+    dispatch('setSimVariables', donuSimulateToD3(modelResults as SimulationResponse[]));
+
+    state.variablesAggregate = donuSimulateToD3([aggregateModelResults(modelResults, aggregator)]);
```

Comment:
Why `variablesAggregate` do not have a `dispatch()` like `variables`?

Reply:
there is no `Action` for setting the `variablesAggregate` property since I did not intend the `variablesAggregate` to be set independently of the fetchModelResults. therefore `dispatch()` is not necessary here

Reply:
I see, we can keep a loose interpretation of the [Action, Getters, Mutations](https://vuex.vuejs.org/vuex.png).
We should just make sure to not get to strict down the line.